src/app/pages/common/ix-forms/services/ix-validators.service.ts

```diff
@@ -0,0 +1,204 @@
+   * and if failed, applies errMessage as the custom error message.
+   */
+  requiredTrue(errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.requiredTrue(control);
+      if (errors?.required) {
+        errors.required = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.email validator to the control
+   * but applies errMessage as the custom error message
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.email on the control
+   * and if failed, applies errMessage as the custom error message.
+   */
+  email(errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.email(control);
+      if (errors?.email) {
+        errors.email = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.minLength validator to the input
+   * but applies errMessage as the custom error message.
+   * @param minLength The minium length allowed
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.minLength and if failed,
+   * applies errMessage as the custom error message.
+   */
+  minLength(minLength: number, errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.minLength(minLength)(control);
+      if (errors?.minlength) {
+        errors.minlength.message = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.maxLength validator to the input
+   * but applies errMessage as the custom error message.
+   * @param maxLength The maximum length allowed
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.maxLength and if failed,
+   * applies errMessage as the custom error message.
+   */
+  maxLength(maxLength: number, errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.maxLength(maxLength)(control);
+      if (errors?.maxlength) {
+        errors.maxlength.message = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.pattern validator to the input
+   * but applies errMessage as the custom error message.
+   * @param pattern The pattern to check the control value against
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.pattern and if failed,
+   * applies errMessage as the custom error message.
+   */
+  pattern(pattern: string | RegExp, errMessage?: string): ValidatorFn {
```

Comment:
I liked your approach of having default messages in the component. 
Could we have a more generic wrapper instead of wrapping each individual original validator?
Something like:
```
cidr_port: ['', this.ixValidators.withMessage(
  Validators.required, 
  this.translate('Yes, it is really required'),
)],
```